adserver/decisionengine/backends.py

```diff
@@ -393,7 +400,26 @@ def select_flight(self):
             # Choose a flight based on the impressions needed
             flight_range = []
             total_clicks_needed = 0
+
             for flight in possible_flights:
+
+                # Apply niche targeting only when any flight has it.
+                # This is to track whether we should do expensive distance queries.
```

Comment:
I put this here so that we only run it when we hit an eligible flight that has niche targeting, so if eg. only house flights have it, it won't run if there's a paid ad.